sql/moz-fx-data-shared-prod/subscription_platform_derived/stripe_subscriptions_history_v1/query.sql

```diff
@@ -200,44 +200,114 @@ plans AS (
   USING
     (product_id)
 ),
+ca_postal_districts AS (
+  -- https://en.wikipedia.org/wiki/Postal_codes_in_Canada#Components_of_a_postal_code
+  SELECT
+    *
+  FROM
+    UNNEST(
+      ARRAY<STRUCT<district STRING, province_code STRING>>[
+        ("A", "NL"),
+        ("B", "NS"),
+        ("C", "PE"),
+        ("E", "NB"),
+        ("G", "QC"),
+        ("H", "QC"),
+        ("J", "QC"),
+        ("K", "ON"),
+        ("L", "ON"),
+        ("M", "ON"),
+        ("N", "ON"),
+        ("P", "ON"),
+        ("R", "MB"),
+        ("S", "SK"),
+        ("T", "AB"),
+        ("V", "BC"),
+        -- District X covers both NT and NU, so we can't tell which one.
+        ("X", NULL),
+        ("Y", "YT")
+      ]
+    )
+),
```

Comment:
I couldn't find a public BigQuery dataset for Canadian postal codes like I found for US zip codes, so this seemed like the least problematic solution for now given that customer billing addresses are no longer being set from PayPal.

It's annoying that this is necessary in the first place though, since FxA's current MaxMind-based geo-location process should be able to get the state/province and save it along with the country and postal code data it's already saving.  I've asked the FxA team about that, and they've opened [FXA-8807](https://mozilla-hub.atlassian.net/browse/FXA-8807) to investigate that possibilty.